packages/core/src/prompts/multi-select.ts

```diff
@@ -10,42 +9,26 @@ interface MultiSelectOptions<T extends { value: any }> extends PromptOptions<Mul
 export default class MultiSelectPrompt<T extends { value: any }> extends Prompt {
 	options: T[];
 	cursor: number = 0;
-	selectedValues: T[];
 
 	private get _value() {
 		return this.options[this.cursor];
 	}
 
-	private changeValue() {
-		const isValueAlreadySelected = this.selectedValues.some((value) => value === this._value.value);
-		if (isValueAlreadySelected) {
-			this.selectedValues = this.selectedValues.filter((value) => value !== this._value.value);
-		} else {
-			this.selectedValues.push(this._value.value);
-		}
```

Comment:
I don't quite grasp how this ever worked, since `selectedValues` is of type `T[]`, i.e. an array of _options_, not _option values_.

Hence, we're comparing options to option values. Same thing goes in UI component.

Am I missing something or did this never work or coincidentally work due to implicit type casts?

Reply:
@banjo what say you?

Reply:
Yeah, I see what you mean now. I would assume that the type `T[]` for `selectedValues` is wrong from the beginning.

`this.selectedValues` comes from the `initialValue` parameter which is basically an array with strings. Or to be more exact: `initialValue?: T['value'][];`.

So even though `selectedValues` are initialized to the type `T[]`, it never actually had that value. 

A bit strange that TypeScript doesn't warn that something is wrong though?

Reply:
Oh I see now. 

The problem lies in the `T` inference. 

```ts
export default class MultiSelectPrompt<T extends { value: any }>
```

It extends any class that has an object with `value: any`. By basically setting `value` to string (or whatever type `value` can be), you wouldn't be able to make that mistake as TypeScript warns again when `selectedValues` are expected to be `T[]`. 

And that part can be fixed with setting `selectedValues` to `string[]` as well.

So basically this would at least make more sense in how it was used. 

```ts
export default class MultiSelectPrompt<T extends { value: string }> extends Prompt {
	options: T[];
	cursor: number = 0;
	selectedValues: string[];

	// ...
```

Even though that really doesn't matter in this new PR 👍

Reply:
Ah yeah nice catch, I don't think this worked as expected previously. Happy with your updates!

Reply:
Yeah, I think we should go through our usages of evil `any`s and get replace them with generics of wherever possible. `any` really breaks type safety.